lib/web_ui/lib/src/engine/bitmap_canvas.dart

```diff
@@ -327,23 +408,53 @@ class BitmapCanvas extends EngineCanvas {
 
   @override
   void drawOval(ui.Rect rect, SurfacePaintData paint) {
-    _setUpPaint(paint);
-    _canvasPool.drawOval(rect, paint.style);
-    _tearDownPaint();
+    if (_useDomForRendering(paint)) {
+      html.HtmlElement element = _buildDrawRectElement(rect, paint,
+          'draw-oval', _canvasPool._currentTransform);
+      _drawElement(element, ui.Offset(math.min(rect.left, rect.right),
+          math.min(rect.top, rect.bottom)), paint);
+      element.style.borderRadius = '${(rect.width/2.0)}px / ${(rect.height/2.0)}px';
+    } else {
+      _setUpPaint(paint);
+      _canvasPool.drawOval(rect, paint.style);
+      _tearDownPaint();
+    }
   }
 
   @override
   void drawCircle(ui.Offset c, double radius, SurfacePaintData paint) {
-    _setUpPaint(paint);
-    _canvasPool.drawCircle(c, radius, paint.style);
-    _tearDownPaint();
+    ui.Rect rect = ui.Rect.fromCircle(center:c, radius:radius);
+    if (_useDomForRendering(paint)) {
+      html.HtmlElement element = _buildDrawRectElement(rect, paint,
+          'draw-circle', _canvasPool._currentTransform);
+      _drawElement(element, ui.Offset(math.min(rect.left, rect.right),
+          math.min(rect.top, rect.bottom)), paint);
+      element.style.borderRadius = '50%';
+    } else {
+      _setUpPaint(paint);
+      _canvasPool.drawCircle(c, radius, paint.style);
+      _tearDownPaint();
+    }
   }
 
   @override
   void drawPath(ui.Path path, SurfacePaintData paint) {
-    _setUpPaint(paint);
-    _canvasPool.drawPath(path, paint.style);
-    _tearDownPaint();
+    if (_useDomForRendering(paint)) {
+      final Matrix4 transform = _canvasPool._currentTransform;
+      final SurfacePath surfacePath = path as SurfacePath;
+      final ui.Rect pathBounds = surfacePath.getBounds();
+      html.Element svgElm = _pathToSvgElement(surfacePath, paint,
+          '${pathBounds.right}', '${pathBounds.bottom}');
+      svgElm.style
+          ..transform = matrix4ToCssTransform(transform)
```

Comment:
Wouldn't this apply the transform twice? Once here and once in `_drawElement`?

Reply:
Done. Yes, it would apply same one twice on the same element, added check.